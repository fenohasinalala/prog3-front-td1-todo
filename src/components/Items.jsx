import React from "react";
import "../styles.css";

function Items({ itemInfo, actionClick }) {
  return (
    <li key={itemInfo.id} className={itemInfo.isComplet ? "item" : "item2"}>
      {itemInfo.nom}
      {!itemInfo.isComplet ? (
        <input
          type="checkbox"
          //checked={itemInfo.isComplet}
          onChange={(e) => {
            if (e.target.checked) {
              actionClick();
              itemInfo.isComplet = true;
            }
          }}
        />
      ) : (
        ""
      )}
    </li>
  );
}

export default Items;
