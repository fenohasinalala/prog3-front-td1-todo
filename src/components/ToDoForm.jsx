import React from "react";
import { useState } from "react";
import { v4 as uuidv4 } from "uuid";

function ToDoForm({ onSubmitAction }) {
  const [newItem, setNewItem] = useState("");

  const handleChange = (event) => {
    setNewItem(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const nom = newItem;
    const id = uuidv4();
    const isComplet = false;
    //const id = "" + Date.now;

    if (nom.length > 0) {
      const toAdd = { id, nom, isComplet };
      onSubmitAction(toAdd);
      setNewItem("");
    }
  };

  return (
    <form action="submit" onSubmit={handleSubmit}>
      <input
        name="newTodo"
        type="text"
        value={newItem}
        placeholder="Ajouter un Item"
        onChange={handleChange}
      />
    </form>
  );
}

export default ToDoForm;
