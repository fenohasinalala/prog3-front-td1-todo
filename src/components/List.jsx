import React from "react";
import Items from "./Items";
import ToDoForm from "./ToDoForm";

function List({ title, itemList, actionCheck, onSumbitAction }) {
  return (
    <div className="list">
      <h1>{title}</h1>
      {title === "TO DO" ? <ToDoForm onSubmitAction={onSumbitAction} /> : ""}
      <ul>
        {itemList.map((item) => {
          return (
            <Items
              itemInfo={item}
              actionClick={() => actionCheck(item.id)}
              key={item.id}
            />
          );
        })}
      </ul>
    </div>
  );
}

export default List;
