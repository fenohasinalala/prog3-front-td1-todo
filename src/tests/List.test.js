import { render, screen } from "@testing-library/react";
import Items from "../components/Items";
import { mocked } from "jest-mock";
import List from "../components/List";
import ToDoForm from "../components/ToDoForm";
import renderer from "react-test-renderer";

jest.mock("../components/Items", () => ({
  __esModule: true,
  default: jest.fn(() => <div>Items</div>),
}));

jest.mock("../components/ToDoForm", () => ({
  __esModule: true,
  default: jest.fn(() => <div>ToDoForm</div>),
}));

const mockItems = mocked(Items);
const mockToDoForm = mocked(ToDoForm);

const mockToDone = jest.fn();
const mockAddItem = jest.fn();

describe("List", () => {
  test("It should display the input textbox if the title is TO DO", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
    const item2 = { id: 2, nom: "Do test homework", isComplet: false };
    render(
      <List
        title={"TO DO"}
        itemList={[item1, item2]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );

    expect(mockToDoForm).toBeCalledTimes(1);
  });

  test("It should not display the input textbox if the title is different from TO DO", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
    const item2 = { id: 2, nom: "Do test homework", isComplet: true };
    render(
      <List
        title={"DONE"}
        itemList={[item1, item2]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );
    expect(screen.queryByRole("textbox")).toBeNull();
  });

  test("It should display one item", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
    render(
      <List
        title={"DONE"}
        itemList={[item1]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );
    expect(mockItems).toHaveBeenCalledTimes(1);
  });

  test("It should display many items", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
    const item2 = { id: 2, nom: "Do test homework", isComplet: true };
    render(
      <List
        title={"DONE"}
        itemList={[item1, item2]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );
    expect(mockItems).toHaveBeenCalledTimes(2);
  });

  it("renders list of TO DO", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
    const item2 = { id: 2, nom: "Do test homework", isComplet: false };
    const component = renderer.create(
      <List
        title={"TO DO"}
        itemList={[item1, item2]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders list of DONE", () => {
    const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
    const item2 = { id: 2, nom: "Do test homework", isComplet: true };
    const component = renderer.create(
      <List
        title={"DONE"}
        itemList={[item1, item2]}
        actionCheck={mockToDone}
        onSumbitAction={mockAddItem}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
