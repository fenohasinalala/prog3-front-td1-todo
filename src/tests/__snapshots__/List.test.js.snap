// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders list of DONE 1`] = `
<div
  className="list"
>
  <h1>
    DONE
  </h1>
  <ul />
</div>
`;

exports[`List renders list of TO DO 1`] = `
<div
  className="list"
>
  <h1>
    TO DO
  </h1>
  <ul />
</div>
`;
